src/ui/viewmodels/MemoryViewerViewModel.cpp

```diff
@@ -1314,6 +1317,33 @@ void MemoryViewerViewModel::RenderHeader()
     }
 }
 
+void MemoryViewerViewModel::SaveToMemViewHistory()
+{
+    if (*s_iterMemViewHistoryIndex != GetAddress())
+    {
+        s_listMemViewHistory.erase(s_listMemViewHistory.begin(), s_iterMemViewHistoryIndex);
+        if (s_listMemViewHistory.size() > MEMVIEW_HISTORY_MAX_SIZE)
+            s_listMemViewHistory.pop_back();
```

Comment:
I think you might have an off-by-one error here. If `size() > MEMVIEW_HISTORY_MAX_SIZE`, that means there's 51 items in the list. Even if you pop the 51st item, you're about to put another 51st back on.